applications/sdp/mspi/src/hrt/hrt.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include "hrt.h"
+#include <hal/nrf_vpr_csr_vio.h>
+#include <hal/nrf_vpr_csr_vtim.h>
+
+#define TOP 4
+
+void write_single_by_word(volatile uint32_t *data, uint8_t data_len, uint32_t counter_top,
+			  uint8_t word_size, bool ce_enable_state, bool hold_ce)
+{
+	NRFX_ASSERT(word_size > MAX_WORD_SIZE);
```

Comment:
Not sure why is this assert here if in line 45 there is code that does alignment when this assertion is false.

Reply:
removed asserts

Reply:
@masz-nordic it is not true. This assert checks if the word size is greater than 32, alignment is done for word sizes smaller than 32. It should either be added back, or checked before calling this function. It will not work with word sizes greater than 32 due to size of the buffer register.

Reply:
in this case this assert is written wrong, this is correct way:
NRFX_ASSERT(word_size <= MAX_WORD_SIZE);
i will put it back fixed

Reply:
the code was working probably only because asserts were disabled